### PR TITLE
container usage clean up

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

File: src/App.php [hunk 1]
```diff
@@ -46,11 +46,12 @@ public function __construct(ContainerInterface $container = null)
         if ($container === null) {
             $container = Container\PHPDiFactory::buildContainer(Loader::load());
         }
-        $this->container = $container;
 
         if ($container->has('router') == false) {
             throw new Exception('Define router config');
         }
```
Comment (samsonasik) [inline src/App.php:38]: throw early, assign later.

File: src/App.php [hunk 2]
```diff
@@ -70,12 +71,12 @@ public function getContainer()
      */
     private function getDispatcher()
     {
-        $container = $this->container;
-        if (!is_callable($container->get('dispatcher'))) {
+        $dispatcher = $this->container->get('dispatcher');
+        if (!is_callable($dispatcher)) {
```
Comment (samsonasik) [inline src/App.php:61]: assignment for `dispatcher` instead.

File: src/App.php [hunk 3]
```diff
@@ -46,11 +46,12 @@ public function __construct(ContainerInterface $container = null)
         if ($container === null) {
             $container = Container\PHPDiFactory::buildContainer(Loader::load());
         }
-        $this->container = $container;
 
-        if ($container->has('router') == false) {
+        if ($container->has('router') === false) {
             throw new Exception('Define router config');
         }
```
Comment (samsonasik) [inline src/App.php:38]: throw early, assign later.

File: src/App.php [hunk 4]
```diff
@@ -105,7 +104,6 @@ public function run($request = null, $response = null)
 
         $event = new HttpFlowEvent('bootstrap', $request, $response);
 
-        $container = $this->getContainer();
```
Comment (samsonasik) [inline src/App.php:94]: `getContainer()` is only usable if called from outside/from App instance, as only return the `container`, use `$this->container` directly instead.

File: src/App.php [hunk 5]
```diff
@@ -46,11 +46,12 @@ public function __construct(ContainerInterface $container = null)
         if ($container === null) {
             $container = Container\PHPDiFactory::buildContainer(Loader::load());
         }
-        $this->container = $container;
 
-        if ($container->has('router') == false) {
+        if ($container->has('router') === false) {
```
Comment (samsonasik) [inline src/App.php:36]: apply `===` for bool comparison

**Comment by samsonasik** [comment]
rebased.

**Comment by samsonasik** [comment]
Scrutinizer error seems not related.

**Comment by samsonasik** [comment]
scrutinzer now green yay ;)

**Comment by samsonasik** [comment]
merge-able ?

**Comment by gianarb** [comment]
yes thanks! :)

**Pull request merged**